app/controllers/reservations_controller.rb

```diff
@@ -23,45 +23,39 @@ def public_timeline
 
-    @start_at     = params[:start] ? Time.zone.at(params[:start].to_i) : Time.zone.now
-    @start_date   = @start_at.beginning_of_day
+    @start_at = params[:start] ? Time.zone.at(params[:start].to_i) : Time.zone.now
+    @end_at = params[:end] ? Time.zone.at(params[:end].to_i) : @start_at.end_of_day
 
-    @end_at       = params[:end] ? Time.zone.at(params[:end].to_i) : @start_at.end_of_day
+    admin_reservations = @instrument.schedule.admin_reservations.in_range(@start_at, @end_at)
+    user_reservations = @instrument.schedule
+                                   .reservations
+                                   .active
+                                   .in_range(@start_at, @end_at)
+                                   .includes(order_detail: { order: :user })
 
-    @admin_reservations = @instrument.schedule.admin_reservations.in_range(@start_at, @end_at)
+    @reservations = admin_reservations + user_reservations
 
-    @user_reservations = @instrument.schedule
-                                    .reservations
-                                    .active
-                                    .in_range(@start_at, @end_at)
-                                    .includes(order_detail: { order: :user })
+    # We don't need the unavailable hours month view
+    unless month_view?
+      @rules = @instrument.schedule_rules
 
-    @reservations = @admin_reservations + @user_reservations
-
-    @rules        = @instrument.schedule_rules
+      if @instrument.requires_approval? && acting_user && acting_user.cannot_override_restrictions?(@instrument)
+        @rules = @rules.available_to_user(acting_user)
+      end
 
-    # restrict to available if it requires approval and the user
-    # isn't a facility admin
-    if @instrument.requires_approval && acting_user && acting_user.cannot_override_restrictions?(@instrument)
-      @rules = @rules.available_to_user(acting_user)
+      @unavailable = ScheduleRule.unavailable(@rules)
     end
 
-    # We're not using unavailable rules for month view
-    @unavailable = if month_view?
-                     []
-                   else
-                     # build unavailable schedule
-                     ScheduleRule.unavailable(@rules)
-                   end
+    @show_details = params[:with_details] == "true" && (@instrument.show_details? || can?(:administer, Reservation))
 
     respond_to do |format|
-      as_calendar_object_options = { start_date: @start_date, with_details: params[:with_details] }
+      as_calendar_object_options = { start_date: @start_at.beginning_of_day, with_details: @show_details }
       format.js do
-        render json: @reservations.map { |r| r.as_calendar_object(as_calendar_object_options) }.flatten +
-                     @unavailable.map { |r| r.as_calendar_object(as_calendar_object_options) }.flatten
+        render json: @reservations.map { |r| r.as_calendar_object(as_calendar_object_options) } +
```

Comment:
The `Array` is there because `@unavailable` may be nil. The `flatten` is not a typo, but definitely a mistake. Turns out that Reservation's `as_calendar_object` behaves differently than `ScheduleRule`'s. Reservation's always returns a single hash, ScheduleRule returns an array of hashes (this is because a single schedule, say weekdays from 9-5, needs 5 calendar items, one for each Mon-Fri). Let me think about how to improve this.